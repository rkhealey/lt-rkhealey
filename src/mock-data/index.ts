import profile from './profile.png';

export const user = {
  username: '@yourname',
  imageSrc: profile,
};

export const theme = {
  background_color: 'rebeccapurple',
  text_color: 'palegoldenrod',
};

export const links = [
  {
    url: 'http://www.google.com?1',
    text: 'Link One',
  },
  {
    url: 'http://www.google.com?2',
    text: 'Link Two',
  },
  {
    url: 'http://www.google.com?3',
    text: 'Link Three',
  },
  {
    url: 'http://www.google.com?4',
    text: 'Link Four',
  },
  {
    url: 'http://www.google.com?5',
    text: 'Link Five',
  },
];

export const showLinks = {
  title: 'Shows',
  links: [
    {
      date: 'April 01 2019',
      venue: 'The Forum, Melbourne',
      url: 'http://www.songkick.com/1',
      available: true,
    },
    {
      date: 'April 02 2019',
      venue: 'Venue Name, Canberra',
      url: 'http://www.songkick.com/2',
      available: false,
    },
    {
      date: 'April 03 2019',
      venue: 'Venue Name, Sydney',
      url: 'http://www.songkick.com/3',
      available: true,
    },
    {
      date: 'April 04 2019',
      venue: 'Venue Name, Brisbane',
      url: 'http://www.songkick.com/4',
      available: true,
    },
  ],
};

export const musicPlayer = {
  title: 'Music',
  artist: 'John Lennon',
  name: 'Imagine',
  links: [
    {
      id: 'spotify',
      platform: 'Spotify',
      url: 'https://open.spotify.com/track/7pKfPomDEeI4TPT6EOYjn9?si=JVKZYLN8RdqzxIFZ_NwhfQ',
      songId: '7pKfPomDEeI4TPT6EOYjn9',
    },
    {
      id: 'apple-music',
      platform: 'Apple Music',
      url: 'https://open.spotify.com/track/7pKfPomDEeI4TPT6EOYjn9?si=JVKZYLN8RdqzxIFZ_NwhfQ&1',
      songId: '7pKfPomDEeI4TPT6EOYjn9',
    },
    {
      id: 'soundcloud',
      platform: 'Soundcloud',
      url: 'https://open.spotify.com/track/7pKfPomDEeI4TPT6EOYjn9?si=JVKZYLN8RdqzxIFZ_NwhfQ&2',
      songId: '7pKfPomDEeI4TPT6EOYjn9',
    },
    {
      id: 'youtube-music',
      platform: 'Youtube Music',
      url: 'https://open.spotify.com/track/7pKfPomDEeI4TPT6EOYjn9?si=JVKZYLN8RdqzxIFZ_NwhfQ&3',
      songId: '7pKfPomDEeI4TPT6EOYjn9',
    },
  ],
};
