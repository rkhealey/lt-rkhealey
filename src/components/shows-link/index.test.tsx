import React from 'react';
import { shallow } from 'enzyme';
import ShowsLink from '.';

import { theme, showLinks } from '../../mock-data';

describe('Shows Link', () => {
  it('Renders', () => {
    const wrapper = shallow(<ShowsLink theme={theme} {...showLinks} />);
    expect(wrapper).toMatchSnapshot();
  });
});
