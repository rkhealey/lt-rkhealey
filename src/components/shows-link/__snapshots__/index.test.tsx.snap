// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shows Link Renders 1`] = `
<Collapse
  theme={
    Object {
      "background_color": "rebeccapurple",
      "text_color": "palegoldenrod",
    }
  }
  title="Shows"
>
  <styled.div
    bg="grey_100"
    pb={7}
    px={3}
    sx={
      Object {
        "backgroundImage": "url(songkick.svg)",
        "backgroundPosition": "left 50% bottom 32px",
        "backgroundRepeat": "no-repeat",
      }
    }
  >
    <Styled(styled.div)
      alignItems="center"
      justifyContent="space-between"
      key="http://www.songkick.com/1"
      py={3}
      variant="withBorderBottom"
    >
      <styled.div>
        <ForwardRef>
          April 01 2019
        </ForwardRef>
        <ForwardRef
          fontSize={0}
        >
          The Forum, Melbourne
        </ForwardRef>
      </styled.div>
      <styled.div
        sx={
          Object {
            "textAlign": "right",
          }
        }
      >
        <ForwardRef
          as="a"
          href="http://www.songkick.com/1"
          target="_blank"
          variant="transparent"
        >
          <i
            className="material-icons"
          >
            chevron_right
          </i>
        </ForwardRef>
      </styled.div>
    </Styled(styled.div)>
    <Styled(styled.div)
      alignItems="center"
      justifyContent="space-between"
      key="http://www.songkick.com/2"
      py={3}
      variant="withBorderBottom"
    >
      <styled.div>
        <ForwardRef>
          April 02 2019
        </ForwardRef>
        <ForwardRef
          fontSize={0}
        >
          Venue Name, Canberra
        </ForwardRef>
      </styled.div>
      <styled.div
        sx={
          Object {
            "textAlign": "right",
          }
        }
      >
        <ForwardRef
          fontSize={0}
        >
          Sold out
        </ForwardRef>
      </styled.div>
    </Styled(styled.div)>
    <Styled(styled.div)
      alignItems="center"
      justifyContent="space-between"
      key="http://www.songkick.com/3"
      py={3}
      variant="withBorderBottom"
    >
      <styled.div>
        <ForwardRef>
          April 03 2019
        </ForwardRef>
        <ForwardRef
          fontSize={0}
        >
          Venue Name, Sydney
        </ForwardRef>
      </styled.div>
      <styled.div
        sx={
          Object {
            "textAlign": "right",
          }
        }
      >
        <ForwardRef
          as="a"
          href="http://www.songkick.com/3"
          target="_blank"
          variant="transparent"
        >
          <i
            className="material-icons"
          >
            chevron_right
          </i>
        </ForwardRef>
      </styled.div>
    </Styled(styled.div)>
    <Styled(styled.div)
      alignItems="center"
      justifyContent="space-between"
      key="http://www.songkick.com/4"
      py={3}
      variant="withBorderBottom"
    >
      <styled.div>
        <ForwardRef>
          April 04 2019
        </ForwardRef>
        <ForwardRef
          fontSize={0}
        >
          Venue Name, Brisbane
        </ForwardRef>
      </styled.div>
      <styled.div
        sx={
          Object {
            "textAlign": "right",
          }
        }
      >
        <ForwardRef
          as="a"
          href="http://www.songkick.com/4"
          target="_blank"
          variant="transparent"
        >
          <i
            className="material-icons"
          >
            chevron_right
          </i>
        </ForwardRef>
      </styled.div>
    </Styled(styled.div)>
  </styled.div>
</Collapse>
`;
